src/js/components/Grommet/Grommet.js

```diff
@@ -110,10 +112,34 @@ class Grommet extends Component {
     }
   };
 
+  ssrResponsive() {
+    const { userAgent } = this.props;
+    const { theme } = this.state;
+
+    if (userAgent) {
+      const md = new MobileDetect(userAgent);
+      if (md.phone()) {
+        return getDeviceBreakpoint('phone', theme);
+      }
+      if (md.tablet()) {
+        return getDeviceBreakpoint('tablet', theme);
+      }
+      return getDeviceBreakpoint('computer', theme);
+    }
+    return undefined;
+  }
+
   render() {
     const { children, ...rest } = this.props;
     delete rest.theme;
-    const { responsive, theme } = this.state;
+    const { responsive: stateResponsive, theme } = this.state;
+
+    // Value from state should be correct once we resize
+    // On first render we try to guess otherwise set the default as a tablet
+    const responsive =
+      stateResponsive ||
+      this.ssrResponsive() ||
```

Comment:
This is looking good. One last comment though. Can we name this `deviceResponsive`? That will align the terms a bit more.